connection.js

```diff
@@ -67,6 +70,23 @@ function connect(id, password, roomName) {
         connection.addEventListener(
             ConnectionEvents.CONNECTION_FAILED, handleConnectionFailed
         );
+        connection.addEventListener(
+            ConnectionEvents.CONNECTION_FAILED, connectionFailedHandler);
+
+        function connectionFailedHandler (error, errMsg) {
```

Comment:
Yes they are very similar but in order to share the code I should add additional arguments and ifs and then I'll need to bind them. I think the code is going to become complex and ugly.